### PR TITLE
LD_LIBRARY_PATH and PYTHONPATH

### PR DESCRIPTION
**Pull request by diegoferigo**
This PR adds to the `setup.sh` files these environment variables, addressing:

- https://github.com/robotology/robotology-superbuild/issues/72
- https://github.com/robotology/idyntree/issues/441

and closing https://github.com/robotology/robotology-superbuild/issues/28.

This should not be necessary but RPATH is broken for python bindings, at least in iDynTree (https://github.com/robotology/idyntree/issues/241).

I couldn't manage to use the old PR https://github.com/robotology/robotology-superbuild/pull/34 because I force pushed to the source branch while it was still closed, operation which apparently prevents reopening it.

cc @traversaro @anqingd @raffaello-camoriano

### CODE REVIEW

**Comment by diegoferigo** [comment]
I'm actually not sure if this is general enough. I mean, right now `FindPython` probably finds python which is python2 and creates the folder `${ROBOTOLOGY_SUPERBUILD_INSTALL_PREFIX}/lib/python2.7/dist-packages`. I would expect that in a system with python pointing to python3 the folder would be `${ROBOTOLOGY_SUPERBUILD_INSTALL_PREFIX}/lib/python3.x/dist-packages`. Can anyone with such setup test it?

Should we use `FindPythonInterp` to get major and minor release?

File: README.md [hunk 1]
```diff
@@ -32,8 +32,9 @@ Table of Contents
   * [Dependencies-specific documentation](#dependencies-specific-documentation)
     * [Gazebo simulator](#gazebo)
     * [MATLAB](#matlab)
-    * [Octave](#octave) 
-  * [FAQs](#faqs)
+    * [Octave](#octave)
+    * [Python](#python)  
+    * [FAQs](#faqs)
```
Comment (traversaro) [inline README.md]: The indentation of `FAQs` seems wrong.

**Comment by traversaro** [comment]
Can you add the documentation on `LD_LIBRARY_PATH` in https://github.com/robotology/robotology-superbuild#configure-your-environment  and `DYLD_LIBRARY_PATH` in https://github.com/robotology/robotology-superbuild#configure-your-environment-1 ?

**Comment by traversaro** [comment]
>  I would expect that in a system with python pointing to python3 the folder would be ${ROBOTOLOGY_SUPERBUILD_INSTALL_PREFIX}/lib/python3.x/dist-packages. Can anyone with such setup test it?

Not a big expert of Python, but the `python` binary will never point to `python3`, see https://www.python.org/dev/peps/pep-0394/ . Given that some subproject actually have problems with Python3 ( https://github.com/robotology/robot-testing/issues/37 ) I think we can stick to default behavior of supporting Python2 for now.

**Comment by diegoferigo** [comment]
@traversaro Done

> The python binary will never point to python3

Ow, ok. I am used to ArchLinux where `python` points to `python3`. I wrongly thought that on newer ubuntu version that was the intended behaviour as well. Apparently, ArchLinux is one of the only exceptions to that rule :sweat_smile:

**Comment by traversaro** [comment]
>  Apparently, Arch Linux is one of the only exceptions to that rule sweat_smile

From the PEP itself: 
> however, end users should be aware that python refers to python3 on at least Arch Linux (that change is what prompted the creation of this PEP), so python should be used in the shebang line only for scripts that are source compatible with both Python 2 and 3.
: D

File: README.md [hunk 2]
```diff
@@ -324,9 +326,11 @@ To use this binaries and libraries, you should update the necessary environment
 
 Set the environment variable `ROBOTOLOGY_SUPERBUILD_ROOT` so that it points to the  directory where you cloned the robotology-superbuild repository.
 
-Append `$ROBOTOLOGY_SUPERBUILD_ROOT/build/install/bin` to your PATH.
+Append `$ROBOTOLOGY_SUPERBUILD_INSTALL_PREFIX/bin` to your PATH.
```
Comment (traversaro) [inline README.md]: I am a bit confused by all this changes to Windows section of the README, given that `DYLD_LIBRARY_PATH` does not exist on Windows. Did you intended to do those in the macOS section?

Comment (diegoferigo) [reply README.md]: Wrong Ctrl+F. My fault, thanks for catching it

File: README.md [hunk 3]
```diff
@@ -533,13 +537,23 @@ brew install octave
...
 ====
 
 See also YCM documentation for [YCM's FAQs](http://robotology.github.io/ycm/gh-pages/master/manual/ycm-faq.7.html).
 
+## Python
+
+Support for this dependency is enabled by the `ROBOTOLOGY_USES_PYTHON` CMake option.
+
+### Configuration
+Add the `$ROBOTOLOGY_SUPERBUILD_INSTALL_PREFIX/lib/python2.7/dist-packages` to your `PYTHONPATH`. This is done automatically if you source the provided `setup.sh` script.
```
Comment (traversaro) [inline README.md]: Prepend "On Linux or macOS" if you mention the `setup.sh` file (this part of the documentation also applies to Windows).

File: README.md (hunk 2 above)
Comment (diegoferigo) [reply README.md]: This README is huge, it's too easy to get lost in the sections :disappointed_relieved:

File: README.md [hunk 4]
```diff
@@ -533,13 +537,23 @@ brew install octave
...
+
+### Check the installation
+The folder above should contain `_*.so` files which correspond to the generated python bindings. Open a python interpreter and try to import modules.
```
Comment (traversaro) [inline README.md]: Please specify the folder to which you are referring, for example "The folder mentioned in the configuration/previous section". 
Furthermore `_*.so` files are only present in *nix system, so I would probably just mention the `.py` files.

**Comment by traversaro** [comment]
@diegoferigo please ping when you address the review, a single commit does not generate a notification (at least, not a clear one) to the reviewer. 

I also added the commit https://github.com/robotology/robotology-superbuild/pull/82/commits/5caf24e32b556d525c9521aaf33b58efb136162f , merging, thanks!

File: cmake/template/setup.sh.in [hunk 5]
```diff
@@ -22,3 +24,8 @@ export GAZEBO_RESOURCE_PATH=${GAZEBO_RESOURCE_PATH}:${ROBOTOLOGY_SUPERBUILD_INST
 # ROBOTOLOGY_ENABLE_DYNAMICS-specific lines
 export YARP_DATA_DIRS=$YARP_DATA_DIRS:${ROBOTOLOGY_SUPERBUILD_INSTALL_PREFIX}/share/codyco
 @endcmakeif ROBOTOLOGY_ENABLE_DYNAMICS
+
+@cmakeif ROBOTOLOGY_USES_PYTHON
+# Add the python bindings directory to the PYTHON_PATH
+PYTHONPATH=${PYTHONPATH}:${ROBOTOLOGY_SUPERBUILD_INSTALL_PREFIX}/lib/python2.7/dist-packages
```
Comment (HosameldinMohamed) [inline cmake/template/setup.sh.in]: @diegoferigo Sorry, why doesn't this line contains the `export` command? For me When I ran `import yarp` in a Python interpreter it gave me an error. Then I added:
```
export PYTHONPATH=${PYTHONPATH}:${ROBOTOLOGY_SUPERBUILD_INSTALL_PREFIX}/lib/python2.7/dist-packages
```
in `.bashrc` and `import yarp` is accepted by Python now.

Comment (diegoferigo) [reply cmake/template/setup.sh.in]: The `export` should be there, you're right. No clue honestly why it's missing.

Comment (traversaro) [reply cmake/template/setup.sh.in]: I think it was just an error on my side. Thanks for  noticing @HosameldinMohamed .

**Pull request merged**